owner/src/main/java/org/the_chance/honeymart/ui/addCategory/AddCategoryViewModel.kt

```diff
@@ -0,0 +1,151 @@
+package org.the_chance.honeymart.ui.addCategory
+
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.update
+import org.the_chance.honeymart.domain.usecase.AddCategoryUseCase
+import org.the_chance.honeymart.domain.usecase.GetAllCategoriesInMarketUseCase
+import org.the_chance.honeymart.domain.util.ErrorHandler
+import org.the_chance.honeymart.ui.base.BaseViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class AddCategoryViewModel @Inject constructor(
+    private val addToCategoryUseCase: AddCategoryUseCase,
+    private val getAllCategories: GetAllCategoriesInMarketUseCase
+) : BaseViewModel<AddCategoryUIState, AddCategoryUIEffect>(AddCategoryUIState()),
+    AddCategoryInteractionListener {
+
+    override val TAG: String = "addCategory"
+
+    init {
+        getCategoryImages()
+        getAllCategory(8)
```

Comment:
Name not clean why all and need pass id I read the code and knew. I see this name is perfect **getMarketCategories**

Reply:
Explained in meet